perf/perf_stress_ng.py

```diff
@@ -0,0 +1,252 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: kajol Jain<kjain@linux.ibm.com>
+#         Manvanthara Puttashakar <manvanth@linux.vnet.ibm.com>
+
+import os
+import re
+import psutil
+import time
+from avocado.utils import process
+from avocado import Test
+from avocado.utils import process, build, archive
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+def clear_dmesg():
+    process.run("dmesg -C ", sudo=True)
+
+
+def collect_dmesg(object):
+    return process.system_output("dmesg").decode()
+
+
+class Stressng(Test):
+
+    """
+    Perf record with CPU stress
+    :source: git://kernel.ubuntu.com/cking/stress-ng.git
+    :param cpu_per: CPU Percentage to load
+    """
+
+    def setUp(self):
+        smm = SoftwareManager()
+        self.timeout = self.params.get('timeout', default='1')
+        self.cpu_per = self.params.get("cpu_load", default='10')
+        self.profile_dur = int(
+            self.params.get(
+                "profile_duration",
+                default='1'))
+        run_type = self.params.get('type', default='upstream')
+        if run_type == "upstream":
+            asset_url = 'https://github.com/ColinIanKing/stress-ng/archive/master.zip'
```

Comment:
just thinking why we need to have upstream support as stress-ng run as workload which whipped from the distro having verified well so we can eliminate any build issue against stress-ng.  what you say?

Reply:
By default yaml setting is distro.  Since the test is based of stressng workload giving an additional option for the test users to switch between upstream and distro variants of stressng.